File: TravelMateBackend/Models/Entities/Booking.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TravelMateBackend.Models.Entities
{
    public class Booking
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public User User { get; set; }
        public Offer Offer { get; set; }
        public Payment Payment { get; set; }
        public BookingStatus Status { get; set; }
        public DateTime ReservedUntil { get; set; }
        public string SeatNumber { get; set; }
        public SeatType SeatType { get; set; }
        public string PassengerName { get; set; }
        public string PassengerType { get; set; }
        public DateTime CreatedAt { get; set; }


    }
    public enum BookingStatus
    {
        Pending,
        Confirmed,
        Canceled
    }

    public enum SeatType
    {
        Economy,
        Business,
        FirstClass
    }

    public enum PassengerType
    {
        Adult,
        Child,
        Baby
    }
}

File: travelmate-userapp/src/app/booking/model/Booking.ts
import {BookingStatus} from "./booking-status.enum";
import {SeatType} from "./seat-type.enum";
import {PassengerType} from "./passenger-type.enum";

export interface Booking {
  Id: string;
  UserId: string;
  OfferId: string;
  PaymentId: string | null;
  Status: BookingStatus;
  ReservedUntil: string;
  SeatType: SeatType;
  PassengerName: string;
  PassengerType: PassengerType;
  CreatedAt: string;
}
